src/Umbraco.Web/PublishedContentQuery.cs

```diff
@@ -195,20 +195,29 @@ public IEnumerable<PublishedSearchResult> Search(string term, int skip, int take
             // default to max 500 results
             var count = skip == 0 && take == 0 ? 500 : skip + take;
 
-            //set this to the specific culture or to the culture in the request
-            culture = culture ?? _variationContextAccessor.VariationContext.Culture;
-
             ISearchResults results;
-            if (culture.IsNullOrWhiteSpace())
+            if (culture == "*")
             {
+                //search everything
+
                 results = searcher.Search(term, count);
             }
+            else if (culture.IsNullOrWhiteSpace())
+            {
+                //only search invariant
+
+                var qry = searcher.CreateQuery().Field(UmbracoContentIndex.VariesByCultureFieldName, "n"); //must not vary by culture
+                qry = qry.And().ManagedQuery(term);
+                results = qry.Execute(count);
+            }
             else
             {
+                //search only the specified culture
+
                 //get all index fields suffixed with the culture name supplied
-                var cultureFields = umbIndex.GetCultureFields(culture);
+                var cultureFields = umbIndex.GetCultureAndInvariantFields(culture).ToArray();
                 var qry = searcher.CreateQuery().Field(UmbracoContentIndex.VariesByCultureFieldName, "y"); //must vary by culture
```

Comment:
```suggestion
                var qry = searcher.CreateQuery();
```

Sorry github do not support multiline suggestions.. But I think the issue is because we require the `Field(UmbracoContentIndex.VariesByCultureFieldName, "y")`

Reply:
@bergmania hrm 🤔 If we are searching for a culture surely we don't want to return invariant documents, only variant documents?

Reply:
I understand from a programmer perspective, but not from a user perspective. 
From a user perspective, a search for a given culture, is the same as navigating a tree in a given culture. And when I navigate the tree, I see both the invariant and variant items?

Reply:
That's really a fundamental change to how things have been designed though. In that sense, that would really come down to where you start searching from in the tree and what cultures you have assigned there which is not taken into account here at all.